### PR TITLE
Add back fetching of blobs in fetch snapshot tree call

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Refer: https://dev.azure.com/fluidframework/internal/_workitems/edit/2207
We had an issue where if we don't fetch blobs in getSnapshotTree call, then if we somebody want to fetch any blob from that snapshot later on, and there is a summary in between which would immediately delete the previous summary, then that blob would be deleted and we would fail.
Reverting this change: https://github.com/microsoft/FluidFramework/commit/cb14d0a8ef2d8b36f6ec451c62aad060b775268c

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 1]
```diff
@@ -312,14 +312,7 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
...
                                 method = "network";
-                                const options: ISnapshotOptions = { ...hostSnapshotOptions };
-                                // Don't fetch the blobs/deltas if it is not the first call. By default server will add
-                                // blobs and deltas to the response.
-                                if (!this.firstVersionCall) {
-                                    options.blobs = 0;
-                                    options.deltas = 0;
-                                }
-                                retrievedSnapshot = await this.fetchSnapshot(options, scenarioName);
+                                retrievedSnapshot = await this.fetchSnapshot(hostSnapshotOptions, scenarioName);
```
Comment (agarwal-navin) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:315]: There are other changes I see in the snapshot that changed this - https://github.com/microsoft/FluidFramework/pull/11784/files.
For instance, an additional param  `cacheOps` was added to `initializeFromSnapshot`. Should that be removed too?

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:315]: No, that is fine.

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:315]: It is just saying that we don't want to override previous ops fetch.

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts [hunk 2]
```diff
@@ -60,9 +60,7 @@ export async function fetchSnapshot(
     snapshotDownloader: (url: string, fetchOptions: { [index: string]: any; }) => Promise<IOdspResponse<unknown>>,
 ): Promise<ISnapshotContents> {
     const path = `/trees/${versionId}`;
-    // We just want to fetch tree here. So explicitly set to not fetch blobs and deltas
-    // as by default server will send that.
-    const queryParams: ISnapshotOptions = { deltas: 0, blobs: 0 };
+    const queryParams: ISnapshotOptions = { deltas: 0, blobs: 1 };
```
Comment (agarwal-navin) [inline packages/drivers/odsp-driver/src/fetchSnapshot.ts:63]: `fetchSnapshot` used to have an additional parameter `fetchFullSnapshot` which was removed by the PR that had changed this  - https://github.com/microsoft/FluidFramework/pull/11784/files.
Can you please make sure that it doesn't need to be added back.

**Comment by agarwal-navin** [comment]
I would recommend backing out the entire change that caused the issue since there could be other issues because of it - https://github.com/microsoft/FluidFramework/pull/11784/files. Then re-add this safely.

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts (hunk 2 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/fetchSnapshot.ts:63]: Yes, that was not required. It was always set to true.

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts [hunk 3]
```diff
@@ -48,21 +48,26 @@ export enum SnapshotFormatSupportType {
...
     snapshotDownloader: (url: string, fetchOptions: { [index: string]: any; }) => Promise<IOdspResponse<unknown>>,
 ): Promise<ISnapshotContents> {
     const path = `/trees/${versionId}`;
-    // We just want to fetch tree here. So explicitly set to not fetch blobs and deltas
-    // as by default server will send that.
-    const queryParams: ISnapshotOptions = { deltas: 0, blobs: 0 };
+    let queryParams: ISnapshotOptions = {};
+
+    if (fetchFullSnapshot) {
+        queryParams = versionId !== "latest" ? { blobs: 2 } : { deltas: 1, blobs: 2 };
```
Comment (agarwal-navin) [inline packages/drivers/odsp-driver/src/fetchSnapshot.ts:68]: `channels` is not needed here anymore?

**Comment by agarwal-navin** [comment]
Can you please add a link to the PR this change backs out so we preserve that history.

**Review by agarwal-navin — APPROVED** [review]

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts (hunk 3 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/fetchSnapshot.ts:68]: That is not read at server anymore. We were told to remove it.

**Pull request merged**